### ISSUE TITLE
Export to PDF of helper list for admins

### ISSUE THREAD

**Issue by dottorer** [lale-help/lale-help#235]
Administrators should have the ability to export the entire helper list into PDF or XLS. This is required for them to submit them to other groups (public offices) in some cases.

![image](https://cloud.githubusercontent.com/assets/14217285/13903477/1bbb777c-ee7d-11e5-8746-1a730f8db74a.png)

We should offer PDF and XLS export fro mthe helpr list page (see top right). Only for circle admins.

**Comment by NolanChan** [comment]
I'm currently writing up something for PDF's!

**Comment by dottorer** [comment]
Awesome - much wanted feature!

**Comment by dottorer** [comment]
@NolanChan @phillipoertel

I deployed a review app for this and added nolanchan@berkeley.edu to it. The access to review apps is 'admin@lale.help' / 'Example1234'.

Trying out some layout things, I found the following:
![image](https://cloud.githubusercontent.com/assets/14217285/19636375/43845c54-99c9-11e6-9fac-d6c0a5468b2f.png)
 
- I think we need to suppress the first column (which is the helper icon) and start with the name column
- phone number does not layout well yet
- columns could use a little spacing / padding
- the last 2 cols are red which I am unsure of why

Also, I think most of the admins will try to export data first into Excel, then massage it a bit (drop columns, fix up data, etc.) to suit their needs and them export or print it. So CSV would be the most bang for the buck.

But a great start on this feature!

**Comment by NolanChan** [comment]
CSV is much more straightforward so I can write something up for that.
And for the PDF, I think the less columns we include, the better looking it will be. If you let me know which columns to include/exclude, I can write up a quick layout file

**Comment by dottorer** [comment]
@NolanChan  Thanks for looking at CSV, we should include all fields in there.

For PDF, I think we should keep the following:

- Name
- Address (pretty layout with line feeds)
- Phone number (pretty layout in 2 lines if both mobile and land line are given)
- Email
- Member since
- Accredited until
- Last login at (all three with readable date format)

Thanks!

**Comment by NolanChan** [comment]
Could I get an example of what a pretty layout/readable format is? (Or where it exists in the code base? Thanks!)

**Comment by dottorer** [comment]
@NolanChan 
![image](https://cloud.githubusercontent.com/assets/14217285/19721222/a71b872a-9b71-11e6-9223-0b58914576ac.png)

I was only referring to 
1. better cell padding to avoid run-in text
2. for address, use format of multiline with:
*street*,
*city*,
*postal*, *state*, *country*
3. for phone, use 2 lines mobile and landline 
*M: xxx xxxx xxxx*
*H: yyy yyyy yyyy*
4. Email do not wrap in col
5. Dates do never wrap (but if possible have localized date format based on user language setting)

Makes sense? I think overall CSV will be more important, but maybe we cna clean up the PDF a bit this way.

**Comment by dottorer** [comment]
Hi @NolanChan , I was working with @phillipoertel on a few things and he took a quick look at the PDF PR as well. We should be able to merge and deploy without issues when we have the layout done on the reduced number of columns for PDF. Sounds like a plan?

Also, I split up the CSV generation in a separate issue in #423. This way we can take it bitesized.

**Comment by NolanChan** [comment]
Sounds good, I'll post up the new code and picture when I get home from work. I have a working stereotype but i haven't had the time to upload it

**Comment by dottorer** [comment]
@NolanChan Very nice! in a new review app, I tested:

- admin user in English and German can see PDF button and get correct language column headings!
- non-admin user does not get to see PDF button
- layout looks good in english and german
![image](https://cloud.githubusercontent.com/assets/14217285/19882580/ecf64478-a00e-11e6-992c-16ee936f96a2.png)

Only nits:
- name in first column is blue link that goes to nowhere. I think we would have this just be black text.
- when address is not specified, we show "   ,   ,   ,  " in  the field which we could drop for cleanliness if it's easy.

Other than that, perfect new feature!!!

**Comment by NolanChan** [comment]
@dottorer , I'll see how easily these changes are and I'll get back to you.

**Comment by NolanChan** [comment]
Hi @dottorer , sorry it's been so long since a status update. The second and third bullet are pretty easy to implement (I've done so already); the first (of getting the names to be black) is sadly a bit more difficult because I'm not exactly sure how Wicked_PDF plays with regular Rail's CSS files. I haven't had any time to devote to this because of other projects, but I hope to get to it this week.

**Comment by dottorer** [comment]
@NolanChan You're a star - thanks for the updates. I might merge the latest changes in and then we can update again if the linkage can be suppressed.

Once we get this out, I am sure the will scream more for CSV than further PDF enhancements, I would expect. Getting lots of new demo customers and more production customers as well now, so usage increased :-)

**Comment by NolanChan** [comment]
@dottorer I updated my branch with the code changes, but the PR should stay the same. Let me know what you think! I'll look into the CVS option now

**Comment by dottorer** [comment]
@NolanChan I saw that the file name is now nicely formed and the commas are gone, but the first column is still a link in the deployed build of the PR. Same for you?

![image](https://cloud.githubusercontent.com/assets/14217285/20463543/a2a8b672-af35-11e6-90b7-ea2557c3a62a.png)

Let me know if we have all the latest in the branch deployed and I will test, merge and push out for all to use.

**Comment by NolanChan** [comment]
Hey @dottorer , I forgot to take off the link. I just pushed up the newest version where the names are no longer links; I also included the translations. Also, are you still seeing the names as blue? In my localhost, I made sure the names are grey as such:  

<img width="1146" alt="screen shot 2016-11-20 at 6 44 01 pm" src="https://cloud.githubusercontent.com/assets/1487616/20469070/caa370b0-af51-11e6-9332-22d0260c9229.png">
